### PR TITLE
feat: transform dw feature `dockerSave`

### PR DESCRIPTION
**Pull request by petemoore**
Fixes taskcluster/taskcluster#5989.

You can see the docker-worker implementation of this feature [here](https://github.com/taskcluster/taskcluster/blob/main/workers/docker-worker/src/features/docker_save.js#L22-L57).

### CODE REVIEW

**Comment by matt-boris** [comment]
Example task of the `podman save` command working successfully https://community-tc.services.mozilla.com/tasks/OwxjTz7xRdu4KFa7E0JPQw/runs/0.

File: d2g.go [hunk 1]
```diff
@@ -106,22 +106,34 @@ func artifacts(artifacts map[string]dockerworker.Artifact) []genericworker.Artif
...
+		gwArt.Type = "file"
+
+		gwArtifacts = append(gwArtifacts, *gwArt)
+	}
+
 	return gwArtifacts
 }
 
-func command(payload *dockerworker.DockerWorkerPayload, dwImage Image, gwArtifacts []genericworker.Artifact, gwWritableDirectoryCaches []genericworker.WritableDirectoryCache) ([][]string, error) {
+func command(dwPayload *dockerworker.DockerWorkerPayload, dwImage Image, gwArtifacts []genericworker.Artifact, gwWritableDirectoryCaches []genericworker.WritableDirectoryCache) ([][]string, error) {
```
Comment (matt-boris) [inline d2g.go:131]: Renamed to `dwPayload` for consistency throughout the rest of the file and for readability sake. Reader can easily understand further down into the function that it's the docker worker payload that's being referenced.

File: d2g.go [hunk 2]
```diff
@@ -150,31 +169,41 @@ func command(payload *dockerworker.DockerWorkerPayload, dwImage Image, gwArtifac
...
+		// when dockerSave is enabled. That artifact will not be
+		// found in either the docker worker payload or the container
+		// after the podman run command is complete, so no podman cp
+		// command is needed for it.
+		// The image artifact is created after the podman run
+		// command is complete.
+		if _, ok := dwPayload.Artifacts[gwArtifacts[i].Name]; !ok {
+			continue
+		}
+		commands = append(commands, "podman cp '"+containerName+":"+dwPayload.Artifacts[gwArtifacts[i].Name].Path+"' "+gwArtifacts[i].Path)
```
Comment (lotas) [inline d2g.go:206]:
```suggestion
		commands = append(commands, fmt.Sprintf("podman cp '%s:%s' %s", containerName, dwPayload.Artifacts[gwArtifacts[i].Name].Path, gwArtifacts[i].Path))
```

might be a bit easier to read

File: testdata/testcases/decision_task_tests.yml [hunk 3]
```diff
@@ -69,11 +69,15 @@ testSuite:
...
+          - expires: '0001-01-01T00:00:00.000Z'
+            name: public/dockerImage.tar.gz
+            path: image.tar.gz
+            type: file
         command:
           - - bash
             - '-cx'
             - >-
-              podman run --name taskcontainer --privileged --cap-add=SYS_PTRACE -v
+              podman run --name taskcontainer --privileged -v
```
Comment (matt-boris) [inline testdata/testcases/decision_task_tests.yml:80]: `--privileged` and  `--cap-add=SYS_PTRACE` cannot be used together in a real test. The `podman run` command will fail out saying you cannot have them both, so I removed the `allowPtrace` feature for now so the test creates a working command.

**Review by lotas — DISMISSED** [review]
👍

**Pull request merged**